x/ccv/types/ccv.go

```diff
@@ -99,7 +120,45 @@ func (cp ConsumerPacketData) ValidateBasic() (err error) {
 	return
 }
 
+// Convert to bytes while maintaining over the wire compatibility with previous versions.
 func (cp ConsumerPacketData) GetBytes() []byte {
-	bytes := ModuleCdc.MustMarshalJSON(&cp)
+	return cp.ToV1Bytes()
```

Comment:
So consumers always send packets in v1/v2 format, why do we still check for the v3 format on provider?

Reply:
Because the packet contains either a VSCMatured or a SlashPacket inside the same structure.